src/database-controller/deploy/database-controller.yaml.template

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: database-controller-sts
+spec:
+  selector:
+    matchLabels:
+      app: database-controller
+  serviceName: database-controller
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: database-controller
+    spec:
+      serviceAccountName: database-controller-account
+      hostNetwork: true
+      initContainers:
+      - name: database-initializer
+        image: {{ cluster_cfg["cluster"]["docker-registry"]["prefix"] }}database-controller:{{ cluster_cfg["cluster"]["docker-registry"]["tag"] }}
+        imagePullPolicy: Always
+        env:
+        - name: LOG_LEVEL
+          value: "{{ cluster_cfg['database-controller']['log-level'] }}"
+        - name: K8S_CONNECTION_TIMEOUT_SECOND
+          value: "{{ cluster_cfg['database-controller']['k8s-connection-timeout-second'] }}"
+        - name: WRITE_MERGER_CONNECTION_TIMEOUT_SECOND
+          value: "{{ cluster_cfg['database-controller']['write-merger-connection-timeout-second'] }}"
+        - name: RECOVERY_MODE_ENABLED
+{% if cluster_cfg['database-controller']['recovery-mode'] %}
+          value: "true"
+{% else %}
```

Comment:
If recovery-mode=false, but you found table is empty, will you still recover from ApiServer?

Reply:
doc the behavior 

---
In reply to: [466121240](https://github.com/microsoft/pai/pull/4761#discussion_r466121240) [](ancestors = 466121240)

Reply:
Done. The recovery mode is now initializer.